### DISCUSSION TITLE
Identifying the root individual

### DISCUSSION THREAD

**Discussion by Snake69** [FamilySearch/GEDCOM#191] [General]
This is probably a dumb question, but I need to make sure of the answer before venturing into a project.

I've looked everywhere for an answer but I can't find anything concrete.  How do I identify the root individual in a Gedcom?  Is it the first INDI record in the Gedcom?  Is it the individual with the lowest ID number?  Is it something else?

Thanks for any help.

**Comment by tychonievich** [comment]
By "root individual" I assume you mean an initial person to highlight in user interfaces and visualizations such as fan charts that put a single individual in some primary position with others displayed as their relatives? If so, the reason you can't find it is because it is not part of GEDCOM.

There is no standard notion related to "root" in GEDCOM (in any version from 3.0 to 7.0). Indeed in many files such a root is not readily definable, either because no individual is more root than others (e.g. files generated by one-name and one-place studies) or because several are equally "root" (e.g. files generated by a collaboration of multiple family members). The `ANCI` and `DESI` structures do provide some connection between users (`SUBM` records) and research subjects (`INDI` records) but are used in multiple ways and are not generally a "root individual" indicator. Matching `SUBM` and `INDI` `NAME`s might also provide some information, but again is not reliable.

Various tools have used non-standard indicators of "root" status and other UI-specific information, for example via Xref values, INDI ordering, and extension structures; Oddly, even though Personal Ancestral File (PAF) had a "root individual" concept and was developed along with early versions of GEDCOM by the same people, PAF's own technique for indicating root individuals (based on Xref values and record order) was never documented in GEDCOM and by version 5.0 was explicitly called out as not carrying meaning and being open to being changed or discarded during import.

I believe the only robust way to pick a root individual that a particular user will agree with is to ask the user which `INDI` they wish to highlight.

**Reply by Snake69** [reply]
Ok, thanks.  I understand.

A follow-up question ... is there any significance to the first INDI record in a GEDCOM ?  Is it perhaps the starting point from where the GEDCOM was created?

**Reply by tychonievich** [reply]
Record order is not officially significant, but I expect some tools do serialize people in some kind of database-row-id order, suggesting they may be roughly in creation order. That assumes they use a relational database with auto-increment rowids rather than a hash-table-based graph database or some other non-creation-order storage medium and don't use a different order-by term in the export query.

**Reply by Snake69** [reply]
Ok.  Thanks very much for your help.

**Comment by WGroleau** [comment]
There is **no** "root" as far as I'm concerned, at least in transmission (the supposed purpose of GEDCOM).  If I am the "root" of my database, would I still be the root when my third cousin once removed receives my file?

**Comment by Snake69** [comment]
Ok.  Point taken.  Thanks.


On Fri, 20 Jan 2023, WGroleau wrote:

> 
> There is no "root" as far as I'm concerned, at least in transmission (the supposed purpose of GEDCOM). If I am the "root" of my database, would
> I still be the root when my third cousin once removed receives my file?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the
> thread.[AB2TKWHHZACE6RKCW2734XLWTLSX5A5CNFSM57XZ24FKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOABEFQLA.gif] Message
> ID: ***@***.***>
> 
> 
>

-- 
Marshall Lake -- ***@***.*** -- http://www.mlake.net

**Comment by fisharebest** [comment]
Since submitters may also have an INDI record in the GEDCOM file, perhaps there is some merit in allowing a submitter to be linked to an individual?

```
0 @X1@ SUBM
1 NAME John Smith
2 ADDR ...
1 INDI @X2@      <-----
...
0 @X2@ INDI
1 NAME John /Smith/
```

This lets us link genealogy records with real-word identities.



I think there may be a similar requirement to link sources to individuals.  I have hundreds of sources "Personal knowledge of John Smith", where John Smith has a record in my GEDCOM data.

**Reply by webtrees-pesz** [reply]
What's about ALIA?

```
0 @X1@ SUBM
1 NAME John Smith
2 ADDR ...
1 ALIA @X2@      <-----
...
0 @X2@ INDI
1 NAME John /Smith/
```

**Reply by webtrees-pesz** [reply]
> I think there may be a similar requirement to link sources to individuals. 
> I have hundreds of sources "Personal knowledge of John Smith", where John Smith has a record in my GEDCOM data.

Why not using SUBM? The submitter record identifies an individual or organization that contributed information contained in the dataset.

```
0 @X3@ SOUR
1 TITL John Smith
1 SUBM @X1@      <-----
...
0 @X1@ SUBM
1 NAME John  Smith
```

**Comment by WGroleau** [comment]
Perhaps like
```
2 SOUR @P202@
3 PAGE e-mail 2017 OCT 18 "Re: my naturalization"
0 @P202@ INDI
1 NAME John /Tomkiewicz/

```

**Comment by fisharebest** [comment]
I was thinking more like

```
0 @X1@ SOUR
1 AUTH
2 INDI @X2@     <---- The author of this source is this individual in the GEDCOM
```

**Comment by WGroleau** [comment]
```
0 @X1@ SUBM
1 NAME John Smith
1 ALIA @X2@
```
and
```
0 @X1@ SUBM
1 NAME John Smith
1 INDI @X2@
are good ideas.
```
But, why do SUBM and INDI have to be different categories?  Could be: 
```
0 HEAD
…
1 SUBM @X123@
or
0 @P345@ INDI
1 SUBM @X123@
```
(@X123@ is an individual who may or may not be related to others in the DB)

For a source authored by someone in the DB (related or not), maybe:
```
0 @X3@ SOUR
1 TITL John Smith
1 AUTH @Dr-Smith@ 
```
(Dr-Smith might be a professional genealogist you hired).  

But, you might have lots of people giving you information directly, and you don't want one source to have dozens of authors.  That's why I suggest allowing the payload of SOUR to be the xref of an INDI with PAGE being the details, like date, occasion, subject line, etc.
```
1 AUTH
2 INDI @X2@ 
```
is just as good, but
```
1 AUTH @X2@ 
```
is shorter.

**Comment by tychonievich** [comment]
See the discussion in #283 for more on root persons and why we need to support several of them in a single file

**Comment by Snake69** [comment]
Thanks, Luther.


On Thu, 30 Mar 2023, Luther Tychonievich wrote:

> 
> See the discussion in #283 for more on root persons and why we need to support several of them in a single file
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the
> thread.[AB2TKWGXLXQ24TGQURYWFC3W6XUMDA5CNFSM57XZ24FKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOABJ2MNA.gif] Message
> ID: ***@***.***>
> 
> 
>

-- 
Marshall Lake -- ***@***.*** -- http://www.mlake.net
